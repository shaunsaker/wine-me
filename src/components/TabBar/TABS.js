const TABS = [
  {
    name: 'Home',
    iconName: 'home',
    sceneKey: '_home',
  },
  {
    name: 'Profile',
    iconName: 'person',
    sceneKey: '_profile',
  },
];

export default TABS;
