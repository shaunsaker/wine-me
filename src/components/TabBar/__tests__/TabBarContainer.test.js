import React from 'react';
import renderer from 'react-test-renderer';

import { TabBarContainer } from '..';
import TABS from '../TABS';

describe('TabBarContainer', () => {
  const spies = [];
  const dispatch = jest.fn();
  const tab = TABS[0];

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<TabBarContainer />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle isCurrentScene', () => {
      const component = renderer.create(<TabBarContainer scene={tab.sceneKey} />);
      const instance = component.getInstance();

      expect(instance.isCurrentScene(tab.sceneKey)).toEqual(true);
      expect(instance.isCurrentScene({ sceneKey: '_profile' })).toEqual(false);
    });

    it('should handle onTabPress', () => {
      spies[0] = jest.spyOn(TabBarContainer.prototype, 'navigate');
      const component = renderer.create(<TabBarContainer scene={tab.sceneKey} />);
      const instance = component.getInstance();

      instance.onTabPress(tab);

      expect(spies[0]).toHaveBeenCalledWith(tab.sceneKey);
    });

    it('should handle navigate', () => {
      const component = renderer.create(<TabBarContainer scene={tab.sceneKey} />);
      const instance = component.getInstance();

      instance.navigate();
    });
  });

  describe('actions', () => {
    // NOTE: We can trust RN to handle press actions in this case
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
