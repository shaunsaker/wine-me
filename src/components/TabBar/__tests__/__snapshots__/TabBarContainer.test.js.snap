// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarContainer renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#48466D",
      "borderColor": null,
      "borderWidth": 0,
      "elevation": 6,
      "flexDirection": "row",
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 2,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
        "paddingBottom": 7,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 0.70)",
            "fontFamily": "Nunito-Regular",
            "fontSize": 15,
          },
          undefined,
        ]
      }
    >
      Home
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
        "paddingBottom": 7,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 24,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 0.70)",
            "fontFamily": "Nunito-Regular",
            "fontSize": 15,
          },
          undefined,
        ]
      }
    >
      Profile
    </Text>
  </View>
</View>
`;
